src/Lykke.Service.Decred.Api/Controllers/TransactionController.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using Common.Log;
+using Lykke.Service.BlockchainApi.Contract.Common;
+using Lykke.Service.BlockchainApi.Contract.Transactions;
+using Lykke.Service.Decred.Api.Common;
+using Lykke.Service.Decred.Api.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Lykke.Service.Decred.Api.Controllers
+{
+    public class TransactionController : Controller
+    {
+        private readonly ILog _log;
+        private readonly IAddressValidationService _addressValidationService;
+        private readonly IUnsignedTransactionService _txBuilderService;
+        private readonly ITransactionBroadcastService _txBroadcastService;
+
+        public TransactionController(
+            ILog log,
+            IAddressValidationService addressValidationService,
+            IUnsignedTransactionService txBuilderService,
+            ITransactionBroadcastService txBroadcastService)
+        {
+            _log = log;
+            _addressValidationService = addressValidationService;
+            _txBuilderService = txBuilderService;
+            _txBroadcastService = txBroadcastService;
+        }
+
+        [HttpGet("api/capabilities")]
+        public CapabilitiesResponse GetCapabilities()
+        {
+            return new CapabilitiesResponse
+            {
+                AreManyInputsSupported = false,
+                AreManyOutputsSupported = false,
+                IsTransactionsRebuildingSupported = false
+            };
+        }
+
+        [HttpPost("api/transactions/single")]
+        public async Task<IActionResult> BuildSingleTransaction([FromBody] BuildSingleTransactionRequest request)
+        {
+            // Do not scale the fee
+            const int feeFactor = 1;
+            return await BuildTxInternal(request, feeFactor);
+        }
+
+        private async Task<IActionResult> BuildTxInternal(BuildSingleTransactionRequest request, decimal feeFactor)
+        {
+            _addressValidationService.AssertValid(request?.FromAddress);
+            _addressValidationService.AssertValid(request?.ToAddress);
+
+            try
+            {
+                var response = await _txBuilderService.BuildSingleTransactionAsync(request, feeFactor);
+                return Json(response);
+            }
+
+            catch (BusinessException exception) when (exception.Reason == ErrorReason.AmountTooSmall)
+            {
+                Response.StatusCode = (int) HttpStatusCode.Conflict;
+                return Json(new
+                {
+                    errorCode = "amountIsTooSmall",
+                    transactionContext = (string) null
+                });
+            }
+
+            catch (BusinessException exception) when (exception.Reason == ErrorReason.NotEnoughBalance)
+            {
+                Response.StatusCode = (int) HttpStatusCode.Conflict;
+                return Json(new
+                {
+                    errorCode = "notEnoughBalance",
```

Comment:
Use [this](https://github.com/LykkeCity/Lykke.Service.BlockchainApi.Client/blob/master/src/Lykke.Service.BlockchainApi.Contract/BlockchainErrorCode.cs) enum for error codes